src/software/ai/hl/stp/play/shoot_or_pass_play.cpp

```diff
@@ -59,83 +59,184 @@ void ShootOrPassPlay::getNextTactics(TacticCoroutine::push_type &yield,
 
+    auto cherry_pick_tactic_1 = std::make_shared<MoveTactic>(false);
+    auto cherry_pick_tactic_2 = std::make_shared<MoveTactic>(false);
+
     auto pitch_division =
         std::make_shared<const EighteenZonePitchDivision>(world.field());
 
+
     PassGenerator<EighteenZoneId> pass_generator(pitch_division,
                                                  play_config->getPassingConfig());
 
-    PassWithRating best_pass_and_score_so_far = attemptToShootWhileLookingForAPass(
-        yield, crease_defender_tactics, attacker, world);
+//    PassWithRating best_pass_and_score_so_far = attemptToShootWhileLookingForAPass(
+//        yield, crease_defender_tactics, attacker, world);
 
-    // If the shoot tactic has not finished, then we need to pass, otherwise we are
-    // done this play
-    if (!attacker->done())
-    {
-        // Commit to a pass
-        Pass pass = best_pass_and_score_so_far.pass;
-
-        LOG(DEBUG) << "Committing to pass: " << best_pass_and_score_so_far.pass;
-        LOG(DEBUG) << "Score of pass we committed to: "
-                   << best_pass_and_score_so_far.rating;
-
-        // Perform the pass and wait until the receiver is finished
-        auto receiver = std::make_shared<ReceiverTactic>(
-            world.field(), world.friendlyTeam(), world.enemyTeam(), pass, world.ball(),
+    auto receiver = std::make_shared<ReceiverTactic>(
+            world.field(), world.friendlyTeam(), world.enemyTeam(), Pass(Point(0, 0), Point(0, 0), 0), world.ball(),
             false);
 
-        auto pass_eval = pass_generator.generatePassEvaluation(world);
+    std::optional<PassWithRating> current_committed_pass = std::nullopt;
+
+    do {
+        auto pass_eval    = pass_generator.generatePassEvaluation(world);
+        auto best_pass_and_score_so_far = pass_eval.getBestPassOnField();
+        auto ranked_zones = pass_eval.rankZonesForReceiving(world, world.ball().position());
+
+//        std::vector<EighteenZoneId> bad_zones = {EighteenZoneId::ZONE_10, EighteenZoneId::ZONE_11,EighteenZoneId::ZONE_12,EighteenZoneId::ZONE_13,EighteenZoneId::ZONE_14,EighteenZoneId::ZONE_15};
+//        for(size_t i = 1; i < ranked_zones.size(); i++) {
+//            if(std::find(bad_zones.begin(), bad_zones.end(), ranked_zones[i]) != bad_zones.end()) {
+//                ranked_zones.erase(ranked_zones.begin() + i);
+//                i--;
+//            }
+//        }
+
+        if (contains(world.field().fieldLines(), world.ball().position())){
+            for(size_t i = 0; i < ranked_zones.size(); i++) {
+                if (contains(pitch_division->getZone(ranked_zones[i]), world.ball().position())) {
+                    ranked_zones.erase(ranked_zones.begin()+i);
+                    i--;
+                    break;
+                }
+            }
+        }
+
+        if (contains(world.field().fieldLines(),attacker->getAssignedRobot()->position())) {
```

Comment:
These `if(contains...)` conditions were us trying to remove adjacent cherry-picker zones.